test/integration/index.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const {test} = require('tap')
+const config = require('../../config.js')
+const nock = require('nock')
+const pkg = require('../../package.json')
+
+// Internal Modules
+const index = require('../../index.js')
+
+const rawEvent = {
+  awslogs: {
+    // eslint-disable-next-line max-len
+    data: 'H4sIAAAAAAAAEzWQQW+DMAyF/wrKmaEkJCbhhjbWCzuBtMNUVSmkNBIQRMKqqep/X6Cb5Ivfs58++45G7ZzqdfMza5Sjt6IpTh9lXReHEsXI3ia9BJnQlHHIhMSEBnmw/WGx6xwcp8Z50M9uN2q/aDUGx2vn/5oYufXs2sXM3tjp3QxeLw7lX6hS47lTz6lTO9i1uynfXkOMe5lsp9Fxzyy/9eS3hTsyXYhOGVCaEsBSgsyEYBkGzrDMAIMQlAq+gQIQSjFhBFgqJOUMAog34WAfoFFOOM8kA0Y5SSH+f0SIb67GRaHq/baosn1UmUlHF7tErxvk5wa56b2Z+iRJ0OP4+AWj9ITzSgEAAA=='
+  }
+}
+
+const eventData = {
+  messageType: 'DATA_MESSAGE'
+, owner: '123456789012'
+, logGroup: 'sampleGroup'
+, logStream: 'testStream'
+, subscriptionFilters: ['LambdaStream_cloudwatchlogs-node']
+, logEvents: [{
+    id: '34622316099697884706540976068822859012661220141643892546'
+  , timestamp: 1557946425136
+  , message: 'This is Sample Log Line for CloudWatch Logging...'
+  }]
+}
+
+const expectedReqBody = [
+  {
+    timestamp: 1557946425136
+  , file: 'testStream'
+  , meta: {
+      owner: '123456789012'
+    , filters: [
+        'LambdaStream_cloudwatchlogs-node'
+      ]
+    , event: {
+        type: 'DATA_MESSAGE'
+      , id: '34622316099697884706540976068822859012661220141643892546'
+      }
+    , log: {
+        group: 'sampleGroup'
+      , stream: 'testStream'
+      }
+    }
+  , line: 'This is Sample Log Line for CloudWatch Logging...'
+  }
+]
+
+test('integration tests', async (t) => {
+  const URL = 'https://ingest.pipeline.com'
+  const KEY = 'abc123'
+  process.env.PIPELINE_KEY = KEY
+  config.set('pipeline-key', KEY)
+  process.env.PIPELINE_URL = URL
+  config.set('pipeline-url', URL)
```

Comment:
Do we need to set the `process.env` variables when the config is being set?

Reply:
I wasn't sure how to get it to not yell at me for having the required in the config.js.  This was the only way I could do that.

Reply:
you'll want to move the `process.env.<variable>` statements to the top of the file right under `use strict` or you can use `config.set()` here

Reply:
I tried config.set() _only_ and I get this error:
` FAIL ​ test/integration/index.js 2 failed of 2 21.534ms
 ✖ Missing required ENV VAR "PIPELINE_URL" of type "string"
 ✖ Missing required ENV VAR "PIPELINE_URL" of type "string"`

Reply:
So I will just move it up to the top, @matt-march

Reply:
If you move the `process.env` setting before the `const config = require('../../config.js')`, are you able to remove the `config.set` calls?

Reply:
No, because then the gets in the real code return null.  I could remove the required in `config.js` and check for it in the code...but that seems silly when I get the feature from `@logdna/env-config`.

Reply:
resolved via Mike's help.